ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -52,17 +52,20 @@
   private final Producer<CommandId, Command> commandProducer;
   private final CommandIdAssigner commandIdAssigner;
   private final AtomicBoolean closed;
+  private final StatementExecutor statementExecutor;
 
   public CommandStore(
       final String commandTopic,
       final Consumer<CommandId, Command> commandConsumer,
       final Producer<CommandId, Command> commandProducer,
-      final CommandIdAssigner commandIdAssigner
+      final CommandIdAssigner commandIdAssigner,
+      final StatementExecutor statementExecutor
```

Comment:
Rather than pass in `StatementExecutor`, just pass in an interface that does what the `CommandStore` needs, e.g. a `QueuedStatementRegister` or something, with a single method. Or even just a `Consumer<CommandId>`. 

This decouples the code more.